target/linux/ath79/image/generic-tp-link.mk

```diff
@@ -135,6 +135,20 @@ define Device/tplink_cpe210-v3
 endef
 TARGET_DEVICES += tplink_cpe210-v3
 
+define Device/tplink_cpe610-v1
+  $(Device/tplink-safeloader)
+  ATH_SOC := ar9344
+  IMAGE_SIZE := 7680k
+  DEVICE_TITLE := TP-Link CPE610 v1
+  TPLINK_BOARD_ID := CPE610V1
+  LOADER_TYPE := elf
+  LOADER_FLASH_OFFS := 0x43000
+  COMPILE := loader-$(1).elf
+  COMPILE/loader-$(1).elf := loader-okli-compile
+  KERNEL := kernel-bin | append-dtb | lzma | uImage lzma -M 0x4f4b4c49 | loader-okli $(1) 12288
```

Comment:
Here I am not sure whether this will work for dynamic partitioning. WITHOUT okli you find the following two options:
KERNEL := kernel-bin | append-dtb | lzma | tplink-v1-header -O
KERNEL := kernel-bin | append-dtb | lzma | uImageArcher lzma

Reply:
I tried it and it gets stuck in a bootloop

Reply:
I will soon (maybe on Monday) receive a CPE210v1 and will then give it a try myself.